### PR TITLE
[GNA] Fix the case when one of the layer outputs is connected to memory

### PR DESCRIPTION
**Pull request by admitriev-gna**
Added fix multiple output with one go to memory where layers were linked to multiple identity layers.
Added tests.

### CODE REVIEW

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 1]
```diff
@@ -145,10 +145,9 @@ static std::vector<CNNLayerPtr> getCandidatesForIdentityInsertion(const CNNLayer
         // for  mul if we have 4-4 - there 2 options
         //          option 1 both inputs came from single outdata  - we will insert 1 identity  to just convert single input into 2 bytes
         //          option 2 each input came from it's own outdata - we need to insert 2 identities activations to convert both and feed weights and inputs
-
+        // TODO: skip non functional
```
Comment (pavel-rodionov) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:148]: `PrevFunctionalLayer` returns non-function layers. So why this Todo is here?

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 2]
```diff
@@ -194,12 +192,91 @@ static std::vector<CNNLayerPtr> getCandidatesForIdentityInsertion(const CNNLayer
...
-        prevLayers.push_back(CNNNetPrevLayer(l, 0));
+        // identity might be already attached to same  layer output but hidden by non functional childs
+        // start iterating all connections
+
+        // TODO: need to have generic function that creates slice of the graph : starting from given layer
+        //  and skipped all non functional - ending up into functional one
+        std::list<CNNLayerPtr> currentSet;
+        std::vector<CNNLayerPtr> resultSet;
+
+        auto & start = prevLayer->outData[outDataIdx]->getInputTo();
```
Comment (esmirno) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:228]: this function should only give a candidate for identities. Modification of topology should be incorporated into dedicated pass.

File: inference-engine/tests/functional/plugin/shared/src/subgraph_tests/multioutput_eltwise_squeeze_eltwise.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include <string>
+#include <vector>
+#include <memory>
+#include <debug.h>
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "subgraph_tests/multioutput_eltwise_squeeze_eltwise.hpp"
+
+namespace LayerTestsDefinitions {
```
Comment (esmirno) [inline inference-engine/tests/functional/plugin/shared/src/subgraph_tests/multioutput_eltwise_squeeze_eltwise.cpp:14]: current code as i remember lacks of correct handling in certain cases, if you didt add case that break it, worth creating different test. Please investigate that more deeply

**Comment by dorloff** [comment]
I propose to fix some typos before merging it into master.

File: inference-engine/src/gna_plugin/gna_graph_tools.hpp [hunk 4]
```diff
@@ -176,6 +176,63 @@ inline std::pair<InferenceEngine::CNNLayerPtr, int>  CNNNetCheckNextLayerSkipCer
     return CNNNetCheckNextLayerSkipCertain(outLayer->second, 0, 0, bOnlyCheck, shouldSkip);
 }
 
+/**
+ * @brief return all layers reacheable from giver one
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/gna_graph_tools.hpp:180]: return all layers reachable from a given one

File: inference-engine/src/gna_plugin/gna_graph_tools.hpp [hunk 5]
```diff
@@ -473,6 +530,30 @@ inline void CNNNetworkInsertLayer(CNNLayerPtr after,
...
+            auto prevPrevLayers = CNNNetGetPrevLayersSkip(prevLayer, acceptanceCriteria);
+            prevLayers.insert(prevLayers.end(), prevPrevLayers.begin(), prevPrevLayers.end());
+        }
+    }
+
+    return prevLayers;
+}
+
 /**
  * @brief remove givven layer from topology, currently only layers with one input data and one output data supported
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/gna_graph_tools.hpp:558]: given

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 6]
```diff
@@ -373,6 +373,7 @@ void GNAPlugin::LoadNetwork(ICNNNetwork &network) {
         passes->registerPass<InsertDiagonalLayerPass>();
         passes->registerPass<HandleMultipleActivationsForTheLayerPass>();
         passes->registerPass<SubstituteScaleShiftBroadCastPass>();
+        passes->registerPass<FuseMultipleIdentiesPass>();
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:376]: Identities, not Identies

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 7]
```diff
@@ -1084,6 +1059,78 @@ void RemoveConstPass::run() {
...
+            if (LayerInfo(res).isIdentity()) {
+                alreadyIdentity = res;
+                break;
+            }
+        }
+        if (!alreadyIdentity) {
+            continue;
+        } else {
+            // just figure out how to connect to that identity already
+            // 1nd stage - disconnect given layer from previous
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:1115]: 1st? 2nd?

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 8]
```diff
@@ -1084,6 +1059,78 @@ void RemoveConstPass::run() {
...
+        for (auto &&res : resultSet) {
+            if (LayerInfo(res).isIdentity()) {
+                alreadyIdentity = res;
+                break;
+            }
+        }
+        if (!alreadyIdentity) {
+            continue;
+        } else {
+            // just figure out how to connect to that identity already
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:1114]: to that "already identity"

**Comment by dorloff** [comment]
@admitriev-gna, please rebase

**Pull request merged**